### ISSUE TITLE
error: Cannot find module "./hooks" from "[object Object]"

### ISSUE THREAD

**Issue by diavrank** [oven-sh/bun#6374]
### What version of Bun is running?

1.0.5+35109160ca5d439116bedeb3302ec3745e2895d5

### What platform is your computer?

Darwin 22.6.0 arm64 arm

### What steps can reproduce the bug?

Load a nest.js app which has these dependencies:

@nestjs/passport
passport-jwt

Script to run:

```shell
"start": "NODE_ENV=localdev nest start --watch --exec \"bun run\"",
```

### What is the expected behavior?

Load the server correctly

### What do you see instead?

```shell
[5:31:59 PM] Starting compilation in watch mode...

[5:32:22 PM] Found 0 errors. Watching for file changes.

error: Cannot find module "./hooks" from "[object Object]"

```

### Additional information

The server startup was working well until 1.0.3, I was waiting for the fix of JWT in https://github.com/oven-sh/bun/issues/2036 , but now I am getting this another error.

**Comment by patrikdevlin** [comment]
I get this same error after installing `dd-trace@^4.16.0` as well.

```
error: Cannot find module "./hooks" from "[object Object]"
```

**Comment by diavrank** [comment]
> I get this same error after installing `dd-trace@^4.16.0` as well.

Actually, I have that dependency as well in version "^4.12.0, so maybe that is the root cause of the problem.

**Comment by dougludlow** [comment]
I get a very similar error in a NestJS app with `dd-trace@3.17.1` installed:

```
error: Cannot find module "./appsec/remote_config" from "[object Object]"
```

Searching for that path in `node_modules` reveals that it indeed originates from `dd-trace`:

```bash
❯ grep "appsec/remote_config" node_modules -R
node_modules/dd-trace/packages/dd-trace/src/proxy.js:const remoteConfig = require('./appsec/remote_config')
```

*Update*

Tried upgrading to the latest `dd-trace@4.17.0` and see the exact same `./hooks` error as above.

**Comment by diavrank** [comment]
Hi, is there any news about this? I think there are a lot of companies using Datadog to monitor their systems. Because of this, I haven't been able to update my Nest js app :/ .

**Comment by Xa3lis** [comment]
> Hi, is there any news about this? I think there are a lot of companies using Datadog to monitor their systems. Because of this, I haven't been able to update my Nest js app :/ .

Same for me, that's the only thing stopping me from fully switch my NestJs app to Bun

**Comment by cheslip** [comment]
It looks like a call to `Module._resolveFilename` is failing in dd-trace. [Link here](https://github.com/DataDog/dd-trace-js/blob/a9c10204122f7d96ea15ddcd2e0215d4fde7d496/packages/dd-trace/src/ritm.js#L54)

Here's a simple reproduction setup:

```
.
└── src
   ├── index.js
   ├── patch.js
   └── test.js
```

```javascript
// src/index.js
require("./patch");
require("./test");

// src/patch.js
const Module = require("module");

const origRequire = Module.prototype.require;

Module.prototype.require = function (request) {
  console.log("Request:", request);
  const filename = Module._resolveFilename(request, this);
  console.log("Resolved:", filename);
  return origRequire.apply(this, arguments);
};

// src/test.js
console.log("success");
```

### bun

```
bun run src/index.js
Request: ./test


error: Cannot find module "./test" from "[object Object]"
```

### node
```
node src/index.js
Request: ./test
Resolved: /.../src/test.js
success
```

**Comment by bitfrost** [comment]
This appears to be fixed as of 1.0.22
` 

u cat main.ts
const tracer = require("dd-trace").init();
➜  u code main.ts
➜  u bun main.ts
Tracer {
  _tracer: NoopTracer {
    _scope: Scope {
      active: [Function: active],
      activate: [Function: activate],
      bind: [Function: bind],
    },
    _span: NoopS`

**Comment by maschwenk** [comment]
❤️ can confirm the same on 1.0.22. Still getting some `dgram` errors but that's a known issue.
